### PR TITLE
Restarting fista

### PR DESCRIPTION
**Pull request by zaccharieramzi**
This solves #34 .

### CODE REVIEW

File: modopt/tests/test_opt.py [hunk 1]
```diff
@@ -119,6 +131,18 @@ def test_forward_backward(self):
         npt.assert_array_equal(self.fb2.x_final, self.data1,
                                err_msg='Incorrect ForwardBackward result.')
 
+        npt.assert_array_equal(self.fb3.x_final, self.data1,
```
Comment (zaccharieramzi) [inline modopt/tests/test_opt.py]: I realized I also needed to check the result of the algorithm for the lazy start strategies, I don't think it's too much of a mess to have it in this PR as well but lmk.

Comment (sfarrens) [reply modopt/tests/test_opt.py]: No problem

**Comment by sfarrens** [comment]
@zaccharieramzi I think you should add yourself as an author of `algorithms.py`  as with this PR you have contributed a lot to this module.

**Comment by zaccharieramzi** [comment]
Done (dk if I did the right way though). I put my CEA address and not the permanent one.

File: modopt/opt/algorithms.py [hunk 2]
```diff
@@ -5,6 +5,7 @@
 This module contains class implementations of various optimisation algoritms.
 
 :Author: Samuel Farrens <samuel.farrens@cea.fr>
```
Comment (sfarrens) [inline modopt/opt/algorithms.py]: Best if you do the following:

```python
:Author: Samuel Farrens <samuel.farrens@cea.fr>, Zaccharie Ramzi <zaccharie.ramzi@cea.fr>
```

**Pull request merged**